ts/screens/modal/RegionServiceModal.tsx

```diff
@@ -0,0 +1,191 @@
+import AppHeader from "../../components/ui/AppHeader";
+import IconFont from "../../components/ui/IconFont";
+import customVariables from "../../theme/variables";
+import { getCurrentLocale } from "../../utils/locale";
+import { showToast } from "../../utils/showToast";
+import {
+  APP_EVENT_HANDLER,
+  AVOID_ZOOM_JS,
+  closeInjectedScript
+} from "../../utils/webview";
+
+type Props = {
+  onModalClose: () => void;
+};
+
+const RegionServiceModal: React.FunctionComponent<Props> = (props: Props) => {
+  const [navigationURI, setNavigationUri] = React.useState("");
+  const [webMessage, setWebMessage] = React.useState("");
+  const [text, setText] = React.useState("");
+  const [loading, setLoading] = React.useState(false);
+  const [success, setSuccess] = React.useState(false);
+  const [error, setError] = React.useState(false);
+
+  React.useEffect(() => {
+    const cookie: Cookie = {
+      name: "test",
+      value: "test",
+      domain: "192.168.1.20", // the domain on which you need to add the cookie, in this case the IP on your machine
+      path: "/",
+      version: "1",
+      expires: "2021-05-30T12:30:00.00-05:00"
+    };
+    /**
+     * the domain on which you need to add the cookie, in this case the IP on your machine
+     * this value and domain value on cookie MUST match
+     * this parameter is considered only by Android.
+     */
+    CookieManager.set("http://192.168.1.20", cookie).catch(_ =>
+      showToast(
+        `Unable to execute cookie authentication for ${cookie.name}`,
+        "danger"
+      )
+    );
+
+    if (navigationURI === "") {
+      setNavigationUri("http://192.168.1.20:3000/test-html");
+    }
+  });
+
+  const showSuccessContent = () => (
+    <Content>
+      <IconFont
+        name={"io-close"}
+        style={{
+          alignSelf: "flex-end"
+        }}
+        onPress={() => setSuccess(false)}
+      />
+      <IconFont
+        name={"io-complete"}
+        size={120}
+        color={customVariables.brandHighlight}
+        style={{
+          alignSelf: "center"
+        }}
+      />
+      <View spacer={true} />
+
+      <Text alignCenter={true}>{`${I18n.t("global.genericThanks")},`}</Text>
+      <Text alignCenter={true} bold={true}>
+        {text}
+      </Text>
+    </Content>
+  );
+
+  const showErrorContent = () => (
+    <Content>
+      <IconFont
+        name={"io-close"}
+        style={{
+          alignSelf: "flex-end"
+        }}
+        onPress={() => setError(false)}
+      />
+      <IconFont
+        name={"io-error"}
+        size={120}
+        color={customVariables.brandDanger}
+        style={{
+          alignSelf: "center"
+        }}
+      />
+      <View spacer={true} />
+
+      <Text alignCenter={true} bold={true}>
+        {text}
+      </Text>
+    </Content>
+  );
+
+  const handleWebviewMessage = (event: WebViewMessageEvent) => {
+    setWebMessage(event.nativeEvent.data);
+
+    const data = JSON.parse(event.nativeEvent.data);
```

Comment:
The codec usage is a best practice in place of raw parsing:

1. [define the type and the codec](https://github.com/pagopa/io-app/blob/b9ba859200159464629aca905bf58f14d625e105/ts/types/MessageCTA.ts) (in this way you will have typed data)
1. [use it to decode the payload](https://github.com/pagopa/io-app/blob/1cc7335a6756f3761b01dca322cdcfa7f4c8712d/ts/utils/messages.ts#L190)

the following code is extremely unsafe: a payload like this, incoming from the web page, causes an app crash

```typescript
const payload = {
                "xy": {
                    "title": "Attenzione",
                    "description": "Sembra che ci sia qualcosa che non va!"
                },
                "zt": {
                    "title": "Warning",
                    "description": "Seems there's something wrong!"
                }
            };
```